src/Polly.Shared/Registry/IPolicyRegistry.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Polly.Registry
+{
+    /// <summary>
+    /// Represents a collection of <see cref="Polly.Policy"/> keyed by <typeparamref name="Key"/>
+    /// </summary>
+    /// <typeparam name="Key">The type of keys in the dictionary</typeparam>
+    /// <typeparam name="Policy">The type of Policy to store. Must have <see cref="Polly.Policy"/> as base class. </typeparam>
+    public interface IPolicyRegistry<Key, Policy> where Policy: Polly.Policy
+    {
+        /// <summary>
+        /// Gets or sets policy with specified Key.
+        /// </summary>
+        /// <param name="key">The key of the policy to get or set.</param>
+        /// <returns>The policy with specified Key.</returns>
+        /// <exception cref="System.ArgumentNullException">Key is null.</exception>
+        /// <exception cref="KeyNotFoundException">The property is retrieved and key is not found.</exception>
```

Comment:
Haha!!! Nice catch. To start with I had copied the comments from `IDictionary` since we had earlier inherited from it. Seems even `IDictionary` has a lot of mistakes. 😁 Fixed.
![image](https://cloud.githubusercontent.com/assets/3169316/23605033/22c0e782-0281-11e7-8d46-152edb361a1c.png)